src/main/java/jimmy/mcgymmy/logic/commands/FindCommand.java

```diff
@@ -17,22 +22,78 @@
     public static final String COMMAND_WORD = "find";
     public static final String SHORT_DESCRIPTION = "Filter the displayed list by a given keyword.";
 
-    private Parameter<FoodContainsKeywordsPredicate> predicateParameter = this.addParameter(
+    private OptionalParameter<FoodContainsKeywordsPredicate> foodPredicateParameter = this.addOptionalParameter(
```

Comment:
I think it will be better to put the food item as a Parameter instead of optional?
If all of them are optional the user can just get away with using just `find ` unless there is some form of a check to ensure not all of them are empty

Reply:
Yeah, this is intended, so that we don't restrict the user from having to input something as name/tag search. For example, if we implemented it as you said, the user could not do `find -d 21-10-2020` to find food items on the 21st of October. Instead, user would have to type in some sort of food name/tag as a mandatory parameter such as `find some food -d 21-10-2020` which would filter out only the food items with name/tag that includes either "some" or "food" + being on that date, making it less intuitive and harder to use the find command meaningfully. 

At the moment there is a check for empty parameters but for your particular concern, `find` on its own will still work, functioning similar to `list`, although I could just as easily change to it returning empty list.

Reply:
Oh ok, I see. I think we can mention that in the DG and no changes will have to be made.

Reply:
I think I only briefly made mention of this under design considerations in DG, do I need to elaborate more on this or can it just be elaborated over in the UG since it concerns the user more?